content/blog/posts/2022-02-01-the-benefits-of-using-nextjs.md

```diff
@@ -0,0 +1,140 @@
+
+```js
+ const ExamplePage = ({ exampleProps }) => {
+   return (
+     //Page components
+   )
+ };
+
+ export async function getServerSideProps() {
+   const res = await fetch('https://...');
+   const exampleProps = await res.json();
+
+   return {
+     props: {
+       exampleProps
+     }
+   };
+ };
+```
+
+Server-side rendering is powerful, and will always generate a page and any corresponding data that is up-to-date. However, this option is slower than static page generation, so the latter should be implemented whenever a page can be successfully generated ahead of a client request.
+
+### SEO Advantages
+
+The pre-rendering options outlined above come with another distinct advantage: SEO improvements. Pre-rendering a page, and its corresponding data, may help your web application's SEO rankings. This is because web crawlers can access the HTML and the data contained on a pre-rendered page; any keywords web crawlers pick up on are then factored into your ranking. Web applications built solely with React do not offer this advantage; since data is fetched after a page is generated, any keywords contained within that page's data will not be accessible by web crawlers.
+
+### File-Based Routing
+
+Another helpful addition Next.js offers is that of file-based page [routing](https://nextjs.org/docs/routing/introduction). The file tree of a Next.js repository is organized so that pages can easily be created under the built-in `pages` directory. Any file with a `.js`, `.jsx`, `.ts`, or `.tsx` extension located in the pages directory is automatically made availabe as a route.
+
+- Any file named `index` is recognized by Next.js to indicate the root of the pages directory, or a sub-directory if located in a sub-folder nested within the pages directory.
+- Next.js allows for both static and dynamic routes. Static routes are explicitly defined by developers, while a dynamic route is defined using brackets, the content of which acts as a placeholder for a dynamic URL segment.
+
+<img width="970" alt="Screen Shot 2022-02-02 at 10 47 56 PM" src="https://user-images.githubusercontent.com/73088654/152288317-7641a935-fda4-470f-a088-31f4c4513c50.png">
+
+`The file tree above displays a dynamic nested route with a single nested index page, and a static nested route with nested dynamic and static pages.`
+
+The built-in file based page routing offered by Next.js eliminates the need to import a routing library typically required when working with a React application. At the same time, the file-based page structure is great for developer experience as it is intuitive, and eliminates the need for explicitly defining routes within the code itself.
+
+<!-- Not sure if I want to include this next section, this blog is already pretty long -->
```

Comment:
I think the next 2 sections (Should I Use Next.js and Key Takeaways) are somewhat repetitive, so  I'd consider consolidating this into 1 section - might also help get the article length where you want it? 

Also, if you're not talking about full stack options in this article, wondering if it's better to exclude that from the bullet list up top? I think a short blurb on this would be a nice to have, though!

Reply:
Yeah these sections are similar but I think are important for readers who may be looking for specific info by title. I may leave as is.

Reply:
Sounds good!